src/diff/children.js

```diff
@@ -302,20 +302,34 @@ function constructNewChildrenArray(newParentVNode, renderResult, oldChildren) {
 				childVNode._flags |= INSERT_VNODE;
 			}
 		} else if (matchingIndex !== skewedIndex) {
-			if (matchingIndex === skewedIndex + 1) {
+			if (matchingIndex == skewedIndex - 1) {
+				skew = matchingIndex - skewedIndex;
+			} else if (matchingIndex == skewedIndex + 1) {
 				skew++;
 			} else if (matchingIndex > skewedIndex) {
+				// Our matched DOM-node is further in the list of children than
+				// where it's at now.
+
+				// When the remaining old children is bigger than the new-children
+				// minus our skewed index we know we are dealing with a shrinking list
+				// we have to increase our skew with the matchedIndex - the skewed index
 				if (remainingOldChildren > newChildrenLength - skewedIndex) {
 					skew += matchingIndex - skewedIndex;
 				} else {
+					// If we have matched all the children just decrease the skew
 					skew--;
 				}
 			} else if (matchingIndex < skewedIndex) {
-				if (matchingIndex == skewedIndex - 1) {
-					skew = matchingIndex - skewedIndex;
+				// Our matched DOM-node is further in the negative way in the list of children
+				// than where it's at now.
+
+				// When the remaining old chiildren is less than the new children
+				// plus our skewed index we know we are dealing with a growing list
+				if (remainingOldChildren < newChildrenLength + skewedIndex) {
+					skew += matchingIndex + skewedIndex;
+				} else {
+					skew = 0;
 				}
-			} else {
```

Comment:
This else branch was unreachable, we check whether we are in a 1 offset first and then lesser or larger